scripts/ado-terraform-nagger-versions.json

```diff
@@ -0,0 +1,36 @@
+{
+  "terraform": {
+    "warn_below": "1.3.7",
+    "error_below": "0.12.0"
+  },
+  "providers": {
+    "registry.terraform.io/chilicat/pkcs12": {
+      "warn_below": "0.0.7",
+      "error_below": "0.0.5"
+    },
+    "registry.terraform.io/dynatrace-oss/dynatrace": {
+      "warn_below": "1.18.1",
+      "error_below": "1.16.0"
+    },
+    "registry.terraform.io/hashicorp/azuread": {
+      "warn_below": "2.33.0",
+      "error_below": "2.19.1"
+    },
+    "registry.terraform.io/hashicorp/azurerm": {
+      "warn_below": "3.42.0",
+      "error_below": "2.99.0"
+    },
+    "registry.terraform.io/integrations/github": {
+      "warn_below": "5.16.0",
+      "error_below": "5.3.0"
+    },
+    "registry.terraform.io/microsoft/azuredevops": {
+      "warn_below": "0.3.0",
+      "error_below": "0.2.0"
+    },
+    "registry.terraform.io/paloaltonetworks/panos": {
+      "warn_below": "1.11.0",
+      "error_below": "1.10.0"
+    }
+  }
+}
```

Comment:
Both Terraform and provider versions are exported to this file, which would need to be managed manually or via automation tools such as updatecli. The `error_below` setting will halt the execution of any pipeline with provider or Terraform versions below those defined in this file. This is for proof-of-concept purposes only, and specific versions at which pipeline execution should be stopped would need to be agreed before merging this PR.